clippy_lints/src/reference.rs

```diff
@@ -48,6 +48,8 @@ impl EarlyLintPass for DerefAddrOf {
     fn check_expr(&mut self, cx: &EarlyContext<'_>, e: &Expr) {
         if let ExprKind::Unary(UnOp::Deref, ref deref_target) = e.kind
             && let ExprKind::AddrOf(_, ref mutability, ref addrof_target) = without_parens(deref_target).kind
+            // NOTE(tesuji): semantics diffences with/out `*&` for array literal `[x, y, ..]`
```

Comment:
Can you link to this- or the other PR so that people can read up on that discussion?

Reply:
also, it'd be good if this could quickly explain what the difference is. like, "avoid warning `*&array` as that may have been intentional to force static-promotion, see #12854 for details" should be good.